pkg/cmd/registry/rule/enable/enable.go

```diff
@@ -0,0 +1,177 @@
+
+	opts := &options{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		Connection: f.Connection,
+		Logger:     f.Logger,
+		localizer:  f.Localizer,
+		Context:    f.Context,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "enable",
+		Short:   f.Localizer.MustLocalize("registry.rule.enable.cmd.description.short"),
+		Long:    f.Localizer.MustLocalize("registry.rule.enable.cmd.description.long"),
+		Example: f.Localizer.MustLocalize("registry.rule.enable.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) (err error) {
+
+			validator := rulecmdutil.Validator{
+				Localizer: opts.localizer,
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			err = validator.ValidateRuleType(opts.ruleType)
+			if err != nil {
+				return err
+			}
+
+			isValid, configs := validator.IsValidRuleConfig(opts.ruleType, opts.config)
+			if !isValid {
+				return opts.localizer.MustLocalizeError("registry.rule.common.error.invalidRuleConfig",
+					localize.NewEntry("RuleType", opts.ruleType),
+					localize.NewEntry("Config", opts.config),
+					localize.NewEntry("ValidConfigList", cmdutil.StringSliceToListStringWithQuotes(configs)),
+				)
+			}
+
+			instanceID, ok := cfg.GetServiceRegistryIdOk()
+			if !ok {
+				return opts.localizer.MustLocalizeError("artifact.cmd.common.error.noServiceRegistrySelected")
+			}
+
+			opts.registryID = instanceID
+
+			return runEnable(opts)
+		},
+	}
+
+	flags := registrycmdutil.NewFlagSet(cmd, f)
+	ruleFlags := rulecmdutil.NewFlagSet(cmd, f)
+
+	flags.AddRegistryInstance(&opts.registryID)
+
+	ruleFlags.AddArtifactID(&opts.artifactID)
+	ruleFlags.AddGroup(&opts.group)
+	ruleFlags.AddConfig(&opts.config)
+	ruleFlags.AddRuleType(&opts.ruleType)
+
+	_ = cmd.MarkFlagRequired("rule-type")
+	_ = cmd.MarkFlagRequired("config")
+
+	return cmd
+
+}
+
+func runEnable(opts *options) error {
+
+	var httpRes *http.Response
+	var newErr error
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	dataAPI, _, err := conn.API().ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+
+	rule := registryinstanceclient.Rule{
+		Config: rulecmdutil.GetMappedConfigValue(opts.config),
+		Type:   rulecmdutil.GetMappedRuleType(opts.ruleType),
+	}
+
+	if opts.artifactID == "" {
+
+		req := dataAPI.AdminApi.CreateGlobalRule(opts.Context)
+
+		req = req.Rule(rule)
+
+		httpRes, newErr = req.Execute()
+		if httpRes != nil {
+			defer httpRes.Body.Close()
+		}
+	} else {
+
+		req := dataAPI.ArtifactRulesApi.CreateArtifactRule(opts.Context, opts.group, opts.artifactID)
+
+		req = req.Rule(rule)
+
+		httpRes, newErr = req.Execute()
+		if httpRes != nil {
+			defer httpRes.Body.Close()
+		}
+
+	}
+
+	if newErr != nil {
+		if httpRes == nil {
+			return newErr
+		}
+
+		operationTmplPair := localize.NewEntry("Operation", "enable")
+		switch httpRes.StatusCode {
```

Comment:
Is this error handling copied from somewhere. We have now error codes for that specifically

Reply:
Ignore me - this is dataplane. Although it is good to have it extracted the same way as we have for control plane apis